cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -954,12 +954,12 @@ private AppStateTable getAppStateTable(StructuredTableContext context) throws Ta
     return new AppStateTable(context);
   }
 
-  private void verifyApplicationExists(NamespaceId namespaceId, String appName) throws ApplicationNotFoundException {
+  private void verifyApplicationExists(NamespaceId namespaceId, String appName)
+    throws ApplicationNotFoundException {
     // Check if app exists
-    ApplicationId appId = namespaceId.app(appName);
-    ApplicationSpecification appSpec = getApplication(appId);
-    if (appSpec == null) {
-      throw new ApplicationNotFoundException(appId);
+    ApplicationMeta latest = getLatest(namespaceId, appName);
```

Comment:
The `getLatest` creates another transaction. From the usage of this `verifyApplicationExists`, we should be able to have the call happens in the same transaction

Reply:
Fixed in commit c0405906d786d2a9ad2dcbc098463bd4655a559f